model/clusters_mgmt/v1/available_regions_inquiry_resource.model

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright (c) 2021 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Manages a specific available regions inquiry.
+resource AvailableRegionsInquiry {
+	// Retrieves the list of available regions of the cloud provider.
+	// IMPORTANT: This list doesn't currently support paging or searching, so the returned
+	// `page` will always be 1 and `size` and `total` will always be the total number of available regions
+	// of the provider.
+	method Search {
```

Comment:
Are you sure you want to name this _Search_? That means that you want to use HTTP requests like this:

```http
POST /api/clusters_mgmt/v1/aws_inquiries/regions/search
````

If that is what you intended and paging isn't supported then I'd suggest to remove the `Page`, `Size` and `Total` parameters.

Reply:
@jhernand we need it as POST to allow payload. We wanted to keep the convention of Page/Size/Total although it's unsupported as parameters to this `search` endpoint, but for consistency of the API. If you think we can remove it, I'm ok with it.

Reply:
I don't have any objection, just wanted to make sure that you intended to use that "/search" path segment.

Reply:
My bad. According to the design doc, the endpoint should not contain "search" suffix. And given this implementation, it's indeed will not contain this suffix. Similar to /available_regions endpoint.